draft/ZEP0010.md

```diff
@@ -0,0 +1,413 @@
+---
+layout: default
+title: ZEP010
+description: This ZEP proposes a new generic extensions field.
+parent: draft ZEPs
+nav_order: 10
+---
+
+# ZEP 10 — Zarr Generic Extensions
+
+Authors:
+
+- [Norman Rzepka](https://github.com/normanrz), scalable minds
+- [Josh Moore](https://github.com/joshmoore), German BioImaging
+
+Status: Draft
+
+Type: Specification
+
+Created: 2025-05-12
+
+## Abstract
+
+This proposal defines a new generic extension point, ``extensions``, to be
+included in the metadata of Zarr v3 arrays and groups. The ``extensions`` field
+provides a consistent mechanism for attaching additional metadata that does not
+fit into existing extension points defined by the core specification. Extension
+entries within this field follow the naming and structure rules established in
+ZEP0009. This mechanism enables third parties to define and share metadata
+extensions without requiring changes to the core specification or introducing
+new top-level keys.
+
+## Introduction
+
+Zarr specification version 3 currently defines four extension points, each
+associated with a specific (array) metadata field. Additional extension points
+may be added by future ZEPs. Until that time, however, third-parties may want
+to add arbitrary extension objects to either arrays or groups. This proposal
+introduces a generic ``extensions`` field that serves as a container for such a
+list of extensions.
+
+## Proposal
+
+To provide for more flexible, immediate, and de-centralized use cases, we
```

Comment:
What I would find persuasive here would be a concrete example of something that people are trying to do with Zarr today that is blocked by the lack of the `extensions` field.

Reply:
I think the example section has a few examples.

Reply:
To make it more explicit, I would take one of these examples
- [xarray's zarr encoding](https://docs.xarray.dev/en/latest/internals/zarr-encoding-spec.html)
- [ome-zarr](https://ngff.openmicroscopy.org/0.5/index.html)
- [chunk accumulation zep](https://github.com/zarr-developers/zeps/blob/main/draft/ZEP0005.md)

introduce in a small number of words how they are using zarr, and then explain why the `extensions` field would solve a real problem for them. This would make it clear that this ZEP is an attempt to solve a real problem that people actually have, unlike the current reference to "flexible, immediate, decentralized use cases".

Reply:
> I am familiar with the examples in the examples section. They are not concrete examples of something that people are trying to do with Zarr today that is blocked by the lack of the `extensions` field.

I disagree. All of these examples are concrete, but simplified, extension proposals that people have been floated by the community.

Reply:
the feedback you are getting is that, from my POV, those examples are not concrete enough, and are too simplified. I personally do not see value in changing the zarr spec to solve a hypothetical, simplified problem. Stating a real problem, and then showing how this proposal will solve that problem, _is_ convincing.